test/QuestTerminalKey.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { expect } from 'chai'
+import { Contract } from 'ethers'
+import { ethers, upgrades } from 'hardhat'
+
+describe('QuestTerminalKey Contract', async () => {
+  let questTerminalKey: Contract,
+    questFactory: Contract,
+    contractOwner: { address: String },
+    royaltyRecipient: { address: String },
+    minterAddress: { address: String },
+    firstAddress: { address: String }
+
+  beforeEach(async () => {
+    ;[contractOwner, royaltyRecipient, minterAddress, firstAddress] = await ethers.getSigners()
+    const QuestTerminalKey = await ethers.getContractFactory('QuestTerminalKey')
+    const QuestFactory = await ethers.getContractFactory('QuestFactory')
+
+    const erc20QuestContract = await ethers.getContractFactory('Quest')
+    const deployedErc20Quest = await erc20QuestContract.deploy()
+
+    questFactory = await upgrades.deployProxy(QuestFactory, [
+      royaltyRecipient.address,
+      firstAddress.address, // really RH Receipt contract but doesnt matter here
+      royaltyRecipient.address,
+      deployedErc20Quest.address,
+      contractOwner.address,
+      firstAddress.address, // really questTerminalKey address but don't have it yet
+    ])
+
+    questTerminalKey = await upgrades.deployProxy(QuestTerminalKey, [
+      royaltyRecipient.address,
+      minterAddress.address,
+      questFactory.address,
+      10,
+      contractOwner.address,
+    ])
+
+    questFactory.setQuestTerminalKeyContract(questTerminalKey.address)
+  })
+
+  describe('Deployment', () => {
+    it('deploys and initializes the the 721 correctly', async () => {
+      expect(await questTerminalKey.symbol()).to.equal('QTK')
+      expect(await questTerminalKey.name()).to.equal('QuestTerminalKey')
+      expect(await questTerminalKey.minterAddress()).to.equal(minterAddress.address)
+      expect(await questTerminalKey.royaltyRecipient()).to.equal(royaltyRecipient.address)
+    })
+  })
+
+  describe('mint', () => {
+    it('mints a token with correct questId', async () => {
+      await questTerminalKey.connect(minterAddress).mint(firstAddress.address, 100, 3)
+      const discount = await questTerminalKey.discounts(1)
+
+      expect(await questTerminalKey.balanceOf(firstAddress.address)).to.eq(1)
```

Comment:
Maybe another simple check to add here is to verify that `ownerOf` `tokenId` 1 is the address of `firstAddress`